src/OpenLoco/Map/TileManager.cpp

```diff
@@ -549,6 +549,64 @@ namespace OpenLoco::Map::TileManager
         _numAnimations = 0;
     }
 
+    // 0x004C5596
+    uint16_t countSurroundingWaterTiles(Map::Pos2 pos)
+    {
+        pos.x = (pos.x & 0xFFE0) - 0xA0;
+        pos.y = (pos.y & 0xFFE0) - 0xA0;
+
+        uint16_t surroundingWaterTiles = 0;
+        for (uint8_t yOffset = 0; yOffset < 11; yOffset++)
+        {
+            for (uint8_t xOffset = 0; xOffset < 11; xOffset++)
+            {
+                auto tile = get(Map::Pos2(pos.x + xOffset, pos.y + yOffset));
```

Comment:
`TilePos2`

Im surprised this isn't causing a divergence as this is just plain wrong.